DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs

```diff
@@ -1,17 +1,153 @@
-﻿using System.Data;
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
 using Dapper;
+using DevEdu.DAL.Enums;
+using DevEdu.DAL.Models;
 
 namespace DevEdu.DAL.Repositories
 {
     public class GroupRepository : BaseRepository, IGroupRepository
     {
+        private const string _groupInsertProcedure = "dbo.Group_Insert";
+        private const string _groupDeleteProcedure = "dbo.Group_Delete";
+        private const string _groupSelectByIdProcedure = "dbo.Group_SelectById";
+        private const string _groupSelectAllProcedure = "dbo.Group_SelectAll";
+        private const string _groupUpdateByIdProcedure = "dbo.Group_UpdateById";
+
+
         private const string _userGroupInsertProcedure = "dbo.User_Group_Insert";
         private const string _userGroupDeleteProcedure = "dbo.Tag_Delete";
         private const string _insertGroupLesson = "dbo.Group_Lesson_Insert";
         private const string _deleteGroupLesson = "dbo.Group_Lesson_Delete";
         private const string _insertGroupMaterial = "dbo.Group_Material_Insert";
         private const string _deleteGroupMaterial = "dbo.Group_Material_Delete";
 
+        public GroupDto AddGroup(GroupDto groupDto)
+        {
+            return _connection.QuerySingle<GroupDto>

```

Comment:
the "Group_Insert" procedure doesn't return GroupDto as I can see
it returns the id of created entity, and this behavior is OK
you've got to call the GetGroup method from BLL with this ID